### PR TITLE
support custom listener options

### PR DESCRIPTION
**Pull request by lovemyliwu**

### CODE REVIEW

**Review by vasco-santos — CHANGES_REQUESTED** [review]
Thanks for this PR @lovemyliwu 
Can you update https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#customizing-transports to document how to configure the listener and add a test?

**Comment by lovemyliwu** [comment]
Sure, but maybe do it later next week

**Comment by vasco-santos** [comment]
> Sure, but maybe do it later next week

Thanks, that is fine! Just @mention me once I can have a look

**Comment by lovemyliwu** [comment]
@vasco-santos please review

File: doc/CONFIGURATION.md [hunk 1]
```diff
@@ -651,6 +651,32 @@ const node = await Libp2p.create({
 })
 ```
 
+Libp2p will auto call `createListener` after initialized transport, if you want custom listener options, you can provide `listenerOptions` in your transport option. For example, `libp2p-webrtc-star` powered by `libp2p-webrtc-peer`, you can custom the ice server(STUN/TURN) config.
```
Comment (vasco-santos) [inline doc/CONFIGURATION.md:654]:
```suggestion
During Libp2p startup, transport listeners will be created for the configured listen multiaddrs.  Some transports support custom listener options and you can set them using the `listenerOptions` in the transport configuration. For example, [libp2p-webrtc-star](https://github.com/libp2p/js-libp2p-webrtc-star) transport listener supports the configuration of its underlying [simple-peer](https://github.com/feross/simple-peer) ice server(STUN/TURN) config as follows:
```

File: doc/CONFIGURATION.md [hunk 2]
```diff
@@ -651,6 +651,32 @@ const node = await Libp2p.create({
...
+
+```js
+const transportKey = WebRTCStar.prototype[Symbol.toStringTag]
+const node = await Libp2p.create({
+  modules: {
+    transport: [WebRTCStar],
+    streamMuxer: [MPLEX],
+    connEncryption: [NOISE]
+  },
```
Comment (vasco-santos) [inline doc/CONFIGURATION.md:663]:
```suggestion
  },
  addresses: {
    listen: ['/dns4/your-wrtc-star.pub/tcp/443/wss/p2p-webrtc-star'] // your webrtc dns multiaddr
  }
```

Comment (vasco-santos) [reply doc/CONFIGURATION.md:663]: We need to specify a listen address so that the transport listener is created

**Review by vasco-santos — CHANGES_REQUESTED** [review]
Overall this looks good! thanks, just need some improvements on the docs

**Comment by lovemyliwu** [comment]
@vasco-santos please review again

**Review by vasco-santos — APPROVED** [review]
LGTM!
THanks @lovemyliwu

**Review by jacobheun — APPROVED** [review]

**Pull request merged**